stream-transformation-tool-service/src/main/java/uk/gov/justice/tools/eventsourcing/transformation/service/RetryStreamOperator.java

```diff
@@ -0,0 +1,85 @@
+package uk.gov.justice.tools.eventsourcing.transformation.service;
+
+
+import static java.lang.String.format;
+
+import uk.gov.justice.services.common.configuration.Value;
+import uk.gov.justice.services.eventsourcing.repository.jdbc.exception.OptimisticLockingRetryException;
+import uk.gov.justice.services.eventsourcing.source.core.EventSourceTransformation;
+import uk.gov.justice.services.eventsourcing.source.core.EventStream;
+import uk.gov.justice.services.eventsourcing.source.core.exception.EventStreamException;
+import uk.gov.justice.services.messaging.JsonEnvelope;
+
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+
+@ApplicationScoped
+public class RetryStreamOperator {
+
+    @Inject
+    private Logger logger;
+
+    @Inject
+    @Value(key = "maxRetryStreamOperation", defaultValue = "10")
+    private long maxRetry;
+
+    @Inject
+    private EventSourceTransformation eventSourceTransformation;
+
+    public void appendWithRetry(final UUID streamId, final EventStream eventStream, final Stream<JsonEnvelope> events) throws EventStreamException {
+        boolean operationCompletedSuccesfully = false;
+        long retryCount = 0L;
+        while (!operationCompletedSuccesfully) {
+            try {
+                eventStream.append(events);
+                logger.info("Appended events to stream with ID - '{}'", streamId);
+                operationCompletedSuccesfully = true;
```

Comment:
It would be possible to encapsulate the actual operation as a Function, and then remove the repeated code for the two methods in this class

Reply:
done